src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisConfigForm.tsx

```diff
@@ -0,0 +1,354 @@
+import { LoadingPropsType } from '@app/Shared/ProgressIndicator';
+import { RecordingAttributes, RecordingOptions } from '@app/Shared/Services/Api.service';
+import { ServiceContext } from '@app/Shared/Services/Services';
+import { AutomatedAnalysisRecordingConfig } from '@app/Shared/Services/Settings.service';
+import { SelectTemplateSelectorForm } from '@app/TemplateSelector/SelectTemplateSelectorForm';
+import { useSubscriptions } from '@app/utils/useSubscriptions';
+import {
+  ActionGroup,
+  Button,
+  Form,
+  FormGroup,
+  FormSection,
+  FormSelect,
+  FormSelectOption,
+  HelperText,
+  HelperTextItem,
+  Split,
+  SplitItem,
+  Text,
+  TextInput,
+  TextVariants,
+  ValidatedOptions,
+} from '@patternfly/react-core';
+import { CogIcon } from '@patternfly/react-icons';
+import * as React from 'react';
+import { concatMap, filter, first, mergeMap, toArray } from 'rxjs';
+import { automatedAnalysisRecordingName } from './AutomatedAnalysisCard';
+
+interface AutomatedAnalysisConfigFormProps {
+  onCreate?: () => void;
+  onSave?: () => void;
+  isSettingsForm: boolean;
+}
+
+export const AutomatedAnalysisConfigForm: React.FunctionComponent<AutomatedAnalysisConfigFormProps> = (props) => {
+  const context = React.useContext(ServiceContext);
+  const addSubscription = useSubscriptions();
+
+  const [templates, setTemplates] = React.useState([] as EventTemplate[]);
+  const [templateName, setTemplateName] = React.useState<string | undefined>(undefined);
+  const [templateType, setTemplateType] = React.useState<TemplateType | undefined>(undefined);
+  const [maxAge, setMaxAge] = React.useState(0);
+  const [maxAgeUnits, setMaxAgeUnits] = React.useState(1);
+  const [maxSize, setMaxSize] = React.useState(0);
+  const [maxSizeUnits, setMaxSizeUnits] = React.useState(1);
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [isSaveLoading, setIsSaveLoading] = React.useState(false);
+  const [showHelperMessage, setShowHelperMessage] = React.useState(false);
+
+  const createButtonLoadingProps = React.useMemo(
+    () =>
+      ({
+        spinnerAriaValueText: 'Creating',
+        spinnerAriaLabel: 'create-active-recording',
+        isLoading: isLoading,
+      } as LoadingPropsType),
+    [isLoading]
+  );
+
+  const saveButtonLoadingProps = React.useMemo(
+    () =>
+      ({
+        spinnerAriaValueText: 'Saving',
+        spinnerAriaLabel: 'save-config-settings',
+        isLoading: isSaveLoading,
+      } as LoadingPropsType),
+    [isSaveLoading]
+  );
+
+  const isFormInvalid: boolean = React.useMemo(() => {
+    return !templateName || !templateType;
+  }, [templateName, templateType]);
+
+  React.useEffect(() => {
+    addSubscription(
+      context.target
+        .target()
+        .pipe(
+          concatMap((target) =>
+            context.api.doGet<EventTemplate[]>(`targets/${encodeURIComponent(target.connectUrl)}/templates`).pipe(
```

Comment:
This setup will close the stream after first call and will not receive notification if target changes. See #687  for more details (check CustomRecordingForm for fixes)

Reply:
To see the issue, just change the target and you will see the template list do not change.

Reply:
I think it does change, I just changed modified the observable piping to filter out "Cryostat" template. Because if someone selects a "Cryostat" template as a default, and then uses it later in a target which does not have that event template, there will be an error.

Reply:
I have refactored out `TargetView` to use `TargetSelect` as of now.

Reply:
Maybe tryyy to select a target the requires authentication(ignore form)/ssl? The template dropdown should appear empty.